### ISSUE TITLE
Active learning simulation does more queries than example

### ISSUE THREAD

**Issue by mikkelam** [JohnLangford/vowpal_wabbit#956]
As title says and as can be seen in my output, using active learning flag along with simulation seems to make more queries than there is examples. Also, not running with --active and --simulation gives the exact same loss?

```
vw -b 28 --ect 3583 -d data.vw --passes 1 -c -k --active --simulation                              
Num weight bits = 28
learning rate = 0.5
initial_t = 0
power_t = 0.5
creating cache_file = data.vw.cache
Reading datafile = data.vw
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.000000 0.000000            1            1.0        1        1       40
0.500000 1.000000            2            2.0        2        1       39
0.750000 1.000000            4            4.0        4        3       39
0.750000 0.750000            8            8.0        7        6       40
0.750000 0.750000           16           16.0       13       12       41
0.781250 0.812500           32           32.0       23       22       38
0.656250 0.531250           64           64.0       33       33       25
0.585938 0.515625          128          128.0       23       15       40
0.578125 0.570312          256          256.0       72     2049       41
0.546875 0.515625          512          512.0      132      132       23
0.536133 0.525391         1024         1024.0       40      202       28
0.544434 0.552734         2048         2048.0      345      344       35
0.538086 0.531738         4096         4096.0      591       93       37
0.534912 0.531738         8192         8192.0       40       40       36
0.504761 0.474609        16384        16384.0       40       39       28
0.496979 0.489197        32768        32768.0      188      189       35
0.472046 0.447113        65536        65536.0      738     1183       18
0.373360 0.274673       131072       131072.0       26       26       27
0.358597 0.343834       262144       262144.0        5        5       42
0.341085 0.323574       524288       524288.0     1838     1838       40
finished run
number of examples per pass = 805573
passes used = 1
weighted example sum = 805573.000000
weighted label sum = 0.000000
average loss = 0.320445
total feature number = 27645642
total queries = 9327136
```

**Comment by JohnLangford** [comment]
ECT is a multiclass-to-binary reduction which calls the binary learner
multiple times per multiclass exmaple.  Active learning only works at the
binary classifier level.  Hence, you are simulating the number of calls of
an oracle required to resolve pairwise comparison labels.  You would need
about 12 binary queries/multiclass with ECT, so you are avoiding a few of
them but not still querying almost everywhere.  To query less, you need to
experiment with --mellowness.

-John
